code/modules/asset_cache/transports/asset_transport.dm

```diff
@@ -40,6 +40,8 @@
 	var/datum/asset_cache_item/ACI = asset
 	if (!istype(ACI))
 		ACI = new(asset_name, asset, file_hash, dmi_file_path)
+		// MD5 calls are slow
+		CHECK_TICK
```

Comment:
Big asset init savings, please confirm sleeping is okay here.

Reply:
are you really just making changes without knowing?

Reply:
I'm not familiar with the system but at a glance there's no reason sleeping should be a problem